app/src/org/commcare/views/TabbedDetailView.java

```diff
@@ -73,11 +73,9 @@ public void setRoot(ViewGroup root) {
 
         mMenu = root.findViewById(R.id.tabbed_detail_menu);
         mViewPager = root.findViewById(R.id.tabbed_detail_pager);
-        mViewPager.setId(AndroidUtil.generateViewId());

```

Comment:
this already has an id set in xml and setting dynamic ids result in fragment not getting re-attached to the activity.

Reply:
Good catch. It seems really strange that the original line was set at all, do you have a sense of what was trying to be accomplished with it? The only reason I can think you'd set the ID manually would be if you intended to have two ViewPagers within the same Activity.

I went back to check on the git blames and it does look like that is explicitly why that change was made:
https://github.com/dimagi/commcare-android/commit/d2eab7ce90d0f6b7d1f30d81fddb8ad9159b9560

The only context where I can think this occurs is with *persistent case tiles* which do have their own view pager. The previous commit from the one I linked above is

https://github.com/dimagi/commcare-android/commit/a5b7f23591a72992788d69060db77c510beaeb67

which does discuss persistent case tiles, and it's in the persistent case tile PR
https://github.com/dimagi/commcare-android/pull/81

I think the key thing to test is loading a case list screen with a view pager, along with a persistent case tile that has its own view pager. I think both end up in the same general view hierarchy, so they need distinct ID's just based on how the views work but it's possible that other refactors have made that unecessary.